src/Build/Logging/BinaryLogger/BuildEventArgsReader.cs

```diff
@@ -33,12 +33,25 @@ public BuildEventArgsReader(BinaryReader binaryReader)
         }
 
         /// <summary>
+        /// Raised when the log reader encounters a binary blob embedded in the stream.
+        /// The arguments include the blob kind and the byte buffer with the contents.
+        /// </summary>
+        internal event Action<BinaryLogRecordKind, byte[]> OnBlobRead;
+
+        /// <summary>
         /// Reads the next log record from the binary reader. If there are no more records, returns null.
         /// </summary>
         public BuildEventArgs Read()
         {
             BinaryLogRecordKind recordKind = (BinaryLogRecordKind)ReadInt32();
 
+            while (IsBlob(recordKind))
```

Comment:
Oh, I see. This is called in a loop from the `BinaryLogReplayEventSource`. I had the wrong impression that this method reads everything. Ignore me :)

---
In reply to: [118305840](https://github.com/Microsoft/msbuild/pull/2109#discussion_r118305840) [](ancestors = 118305840)